Build failure due to CMake "file COPY cannot copy file"

Kodi frequently fails with:
```
[7/1529] No update step for 'dvdcss'
[8/1529] Generating build/xbmc/CompileInfo.cpp, addons/audioencoder.kodi.builtin.aac/addon.xml, addons/audioencoder.kodi.builtin.wma/addon.xml, addons/kodi.binary.global.audioengine/addon.xml, addons/kodi.binary.global.filesystem/addon.xml, addons/kodi.binary.global.general/addon.xml, addons/kodi.binary.global.gui/addon.xml, addons/kodi.binary.global.main/addon.xml, addons/kodi.binary.global.network/addon.xml, addons/kodi.binary.instance.audiodecoder/addon.xml, addons/kodi.binary.instance.audioencoder/addon.xml, addons/kodi.binary.instance.game/addon.xml, addons/kodi.binary.instance.imagedecoder/addon.xml, addons/kodi.binary.instance.inputstream/addon.xml, addons/kodi.binary.instance.peripheral/addon.xml, addons/kodi.binary.instance.pvr/addon.xml, addons/kodi.binary.instance.screensaver/addon.xml, addons/kodi.binary.instance.vfs/addon.xml, addons/kodi.binary.instance.videocodec/addon.xml, addons/kodi.binary.instance.visualization/addon.xml, addons/xbmc.addon/addon.xml
FAILED: build/xbmc/CompileInfo.cpp addons/audioencoder.kodi.builtin.aac/addon.xml addons/audioencoder.kodi.builtin.wma/addon.xml addons/kodi.binary.global.audioengine/addon.xml addons/kodi.binary.global.filesystem/addon.xml addons/kodi.binary.global.general/addon.xml addons/kodi.binary.global.gui/addon.xml addons/kodi.binary.global.main/addon.xml addons/kodi.binary.global.network/addon.xml addons/kodi.binary.instance.audiodecoder/addon.xml addons/kodi.binary.instance.audioencoder/addon.xml addons/kodi.binary.instance.game/addon.xml addons/kodi.binary.instance.imagedecoder/addon.xml addons/kodi.binary.instance.inputstream/addon.xml addons/kodi.binary.instance.peripheral/addon.xml addons/kodi.binary.instance.pvr/addon.xml addons/kodi.binary.instance.screensaver/addon.xml addons/kodi.binary.instance.vfs/addon.xml addons/kodi.binary.instance.videocodec/addon.xml addons/kodi.binary.instance.visualization/addon.xml addons/xbmc.addon/addon.xml 
cd /var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia/.armv8a-libreelec-linux-gnueabi && /var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/toolchain/bin/cmake -DCORE_SOURCE_DIR=/var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia -DCORE_SYSTEM_NAME=linux -DCORE_BUILD_DIR=build -DCMAKE_BINARY_DIR=/var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia/.armv8a-libreelec-linux-gnueabi -DARCH_DEFINES="-D_LINUX -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_MEMFD=1 -DHAVE_MKOSTEMP=1" -DAPP_SCMID=rockchip_18.2rc1-Leia -DAPP_COPYRIGHT_YEARS=2005-2018 -DAPP_BUILD_DATE=20190427 -Dprefix=/var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia/.armv8a-libreelec-linux-gnueabi/build -P /var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia/cmake/scripts/common/GenerateVersionedFiles.cmake
CMake Error at /var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia/cmake/scripts/common/GenerateVersionedFiles.cmake:28 (file):
  file COPY cannot copy file
  "/var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia/addons/xbmc.addon/repository.xsd"
  to
  "/var/lib/jenkins/LE/build2/workspace/box-trn9/build.LibreELEC-RK3328.arm-9.1-devel/kodi-rockchip_18.2rc1-Leia/.armv8a-libreelec-linux-gnueabi/addons/xbmc.addon/repository.xsd".


[9/1529] No patch step for 'dvdcss'
[10/1529] Copying files into build tree
```

LibreELEC is being clobbered by this repeatedly, which is truly annoying when Kodi is just a small part of the entire OS build and a random Kodi failure wastes hours of compute time.

I mentioned this on Team Kodi Slack, and was told the only solution is to repeat the Kodi build, but we don't special case build failures so that's not really an option - either the package (Kodi in this case) should build successfully, or it fails for a valid reason and not an annoying random bug.

I'm opening this issue in the hope that it might be fixed eventually without a hack workaround! Thanks.

Amendment to previous comment:

The file exists in both the source and target:
```
CMake Error at /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel/kodi-18.2rc1-Leia/cmake/scripts/common/GenerateVersionedFiles.cmake:28 (file):
  file COPY cannot copy file
  "/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel/kodi-18.2rc1-Leia/addons/screensaver.xbmc.builtin.dim/resources/language/resource.language.ko_kr/strings.po"
  to
  "/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel/kodi-18.2rc1-Leia/.x86_64-libreelec-linux-gnu/addons/screensaver.xbmc.builtin.dim/resources/language/resource.language.ko_kr/strings.po".
```

Source:
```
ubuntu@buildbox:~/projects$ ls -la /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel/kodi-18.2rc1-Leia/addons/screensaver.xbmc.builtin.dim/resources/language/resource.language.ko_kr/strings.po
-rw-rw-r-- 1 ubuntu ubuntu 698 Apr 29 05:04 /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel/kodi-18.2rc1-Leia/addons/screensaver.xbmc.builtin.dim/resources/language/resource.language.ko_kr/strings.po
```
Target:
```
ubuntu@buildbox:~/projects$ ls -la /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel/kodi-18.2rc1-Leia/.x86_64-libreelec-linux-gnu/addons/screensaver.xbmc.builtin.dim/resources/language/resource.language.ko_kr/strings.po
-rw-rw-r-- 1 ubuntu ubuntu 698 Apr 29 05:04 /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel/kodi-18.2rc1-Leia/.x86_64-libreelec-linux-gnu/addons/screensaver.xbmc.builtin.dim/resources/language/resource.language.ko_kr/strings.po
```

This is a `cmake` based failure: http://ix.io/1OQV
```
make[3]: Leaving directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
Scanning dependencies of target sse4
make[3]: Leaving directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
[  0%] Generating build/xbmc/CompileInfo.cpp, addons/audioencoder.kodi.builtin.aac/addon.xml, addons/audioencoder.kodi.builtin.wma/addon.xml, addons/kodi.binary.global.audioengine/addon.xml, addons/kodi.binary.global.filesystem/addon.xml, addons/kodi.binary.global.general/addon.xml, addons/kodi.binary.global.gui/addon.xml, addons/kodi.binary.global.main/addon.xml, addons/kodi.binary.global.network/addon.xml, addons/kodi.binary.instance.audiodecoder/addon.xml, addons/kodi.binary.instance.audioencoder/addon.xml, addons/kodi.binary.instance.game/addon.xml, addons/kodi.binary.instance.imagedecoder/addon.xml, addons/kodi.binary.instance.inputstream/addon.xml, addons/kodi.binary.instance.peripheral/addon.xml, addons/kodi.binary.instance.pvr/addon.xml, addons/kodi.binary.instance.screensaver/addon.xml, addons/kodi.binary.instance.vfs/addon.xml, addons/kodi.binary.instance.videocodec/addon.xml, addons/kodi.binary.instance.visualization/addon.xml, addons/screensaver.xbmc.builtin.black/addon.xml, addons/screensaver.xbmc.builtin.dim/addon.xml, addons/xbmc.addon/addon.xml
Scanning dependencies of target export-files
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
make[3]: Leaving directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
[  0%] Creating directories for 'libcpluff'
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
[  0%] Creating directories for 'dvdcss'
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
[  0%] Building CXX object build/platform/linux/sse4/CMakeFiles/sse4.dir/CopyFrame.cpp.o
[  0%] Copying files into build tree
[  0%] Built target crossguid
[  0%] Building C object build/cores/dll-loader/exports/CMakeFiles/wrapper.dir/wrapper.c.o
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
CMake Error at /home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/cmake/scripts/common/GenerateVersionedFiles.cmake:28 (file):
  file COPY cannot set modification time on
  "/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu/addons/audioencoder.kodi.builtin.wma/icon.png"


make[3]: *** [CMakeFiles/compileinfo.dir/build.make:87: build/xbmc/CompileInfo.cpp] Error 1
make[3]: Leaving directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.80-devel/kodi-18.3-Leia/.x86_64-libreelec-linux-gnu'
make[2]: *** [CMakeFiles/Makefile2:828: CMakeFiles/compileinfo.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```

Using `ninja` makes this failure much more likely, but the fundamental issue remains even with `cmake` - presumably if this issue can be fixed, then `ninja` would also be fixed.

Another `cmake` fail: http://ix.io/1ORO

```
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/.armv6zk-libreelec-linux-gnueabi'
[  1%] Generating ServiceDescription.h
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/.armv6zk-libreelec-linux-gnueabi'
Scanning dependencies of target retroplayer_messages
make[3]: Leaving directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/.armv6zk-libreelec-linux-gnueabi'
make[3]: Entering directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/.armv6zk-libreelec-linux-gnueabi'
[  1%] Generating AddonModuleXbmcwsgi.i.cpp
CMake Error at /home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/cmake/scripts/common/GenerateVersionedFiles.cmake:28 (file):
  file COPY cannot copy file
  "/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/addons/screensaver.xbmc.builtin.black/icon.png"
  to
  "/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/.armv6zk-libreelec-linux-gnueabi/addons/screensaver.xbmc.builtin.black/icon.png".


make[3]: *** [CMakeFiles/compileinfo.dir/build.make:87: build/xbmc/CompileInfo.cpp] Error 1
make[3]: Leaving directory '/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi.arm-9.80-devel/kodi-18.3-Leia/.armv6zk-libreelec-linux-gnueabi'
```

That's two failures today. I previously tested the `cmake` build last week and had no failures after 1,000 builds. Maybe I'm having rotten luck today, or #16039 is somehow making this failure more likely (ignore the fact the log claims to be building 18.3-Leia as I'm building the latest Kodi 19 master and #16039 is the only change today).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -223,6 +223,7 @@ add_custom_command(OUTPUT ${CORE_BUILD_DIR}/xbmc/CompileInfo.cpp
                                             -Dprefix=${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}
                                             -P ${CMAKE_SOURCE_DIR}/cmake/scripts/common/GenerateVersionedFiles.cmake
                    DEPENDS ${CMAKE_SOURCE_DIR}/version.txt
+                           export-files
                            ${ADDON_XML_DEPENDS}
                            ${CMAKE_SOURCE_DIR}/xbmc/CompileInfo.cpp.in)
 list(APPEND install_data ${ADDON_INSTALL_DATA})

--- a/cmake/modules/FindLibDvd.cmake
+++ b/cmake/modules/FindLibDvd.cmake
@@ -233,6 +233,7 @@ else()
       set(LIBDVD_TARGET_DIR dlls)
     endif()
     copy_file_to_buildtree(${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/libdvd/bin/libdvdnav.dll DIRECTORY ${LIBDVD_TARGET_DIR})
+    add_dependencies(export-files dvdnav)
   endif()
 
   set(LIBDVD_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/libdvd/include)

--- a/cmake/scripts/common/GenerateVersionedFiles.cmake
+++ b/cmake/scripts/common/GenerateVersionedFiles.cmake
@@ -24,9 +24,6 @@ foreach(loop_var ${ADDON_XML_IN_FILE})
   string(REPLACE ${CORE_SOURCE_DIR} ${CMAKE_BINARY_DIR} dest_dir ${source_dir})
   file(MAKE_DIRECTORY ${dest_dir})
 
-  # copy everything except addon.xml.in to build folder
-  file(COPY "${source_dir}" DESTINATION "${CMAKE_BINARY_DIR}/addons" REGEX ".xml.in" EXCLUDE)
-
   configure_file(${source_dir}/addon.xml.in ${dest_dir}/addon.xml @ONLY)
 
   unset(source_dir)

